tests/tas-elasticsearch/src/test/java/org/alfresco/elasticsearch/ElasticsearchTests.java

```diff
@@ -0,0 +1,259 @@
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.testng.AbstractTestNGSpringContextTests;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static java.util.Arrays.asList;
+import static org.testng.Assert.*;
+
+@ContextConfiguration("classpath:alfresco-elasticsearch-context.xml")
+/**
+ * In this test we are verifying end-to-end the indexing and search in Elasticsearch.
+ * In order to test ACLs we created 2 sites and 3 users. 
+ */
+public class ElasticsearchTests extends AbstractTestNGSpringContextTests
+{
+    public static final String INDEX_NAME = "alfresco";
+    public static final String FILE_0_NAME = "test.txt";
+    public static final String FILE_1_NAME = "another.txt";
+    public static final String FILE_2_NAME = "user1.txt";
+    public static final String FILE_3_NAME = "user1Old.txt";
+
+    @Autowired
+    public DataUser dataUser;
+    @Autowired
+    public DataContent dataContent;
+    @Autowired
+    public DataSite dataSite;
+
+    @Autowired
+    protected ServerHealth serverHealth;
+
+    @Autowired
+    protected RestWrapper client;
+
+    private UserModel userSite1;
+    private RestHighLevelClient elasticClient;
+    private SiteModel siteModel;
+    private UserModel userSite2;
+    private UserModel userMultiSite;
+    private FileModel sampleContent;
+
+    @BeforeClass(alwaysRun = true)
+    public void dataPreparation()
+    {
+        serverHealth.assertServerIsOnline();
+
+        elasticClient = new RestHighLevelClient(RestClient.builder(new HttpHost("localhost", 9200, "http")));
+        initIndex();
+
+        userSite1 = dataUser.createRandomTestUser();
+        userSite2 = dataUser.createRandomTestUser();
+        userMultiSite = dataUser.createRandomTestUser();
+
+        siteModel = dataSite.usingUser(userSite1).createPrivateRandomSite();
+        SiteModel otherSiteModel = dataSite.usingUser(userSite2).createPrivateRandomSite();
+
+        dataUser.addUserToSite(userSite2, siteModel, UserRole.SiteContributor);
+        dataUser.addUserToSite(userMultiSite, siteModel, UserRole.SiteContributor);
+        dataUser.addUserToSite(userMultiSite, otherSiteModel, UserRole.SiteContributor);
+
+        sampleContent = createContent(FILE_0_NAME, "This is a test", siteModel, userSite1);
+        createContent(FILE_1_NAME, "This is another file", siteModel, userSite1);
+        createContent(FILE_2_NAME, "This is another file", otherSiteModel, userSite2);
+        createContent(FILE_3_NAME, "This is another file", siteModel, userSite2);
+        //remove the user from site, but he keeps ownership on FILE_3_NAME 
+        dataUser.removeUserFromSite(userSite2, siteModel);
+    }
```

Comment:
A comment would help to understand that userSite1 has access to documents 0, 1 and 3 while userSite2 has access to documents 2 and 3.
Otherwise, reading and understanding the code is required.

Reply:
I added a schema on top of the method, let me know if it is more clear now.

Reply:
Sure, that definitively helps.